src/pages/ProductDetail/DetailContainer.js

```diff
@@ -0,0 +1,176 @@
+              <ProductColorBox
+                picture={'/images/shirt2.png'}
+                onClick={() => {
+                  props.setProductImage('/images/shirt2.png');
+                }}
+              ></ProductColorBox>
+              <ProductColorBox
+                picture={'/images/shirt3.png'}
+                onClick={() => {
+                  props.setProductImage('/images/shirt3.png');
+                }}
+              ></ProductColorBox>
+              <ProductColorBox
+                picture={'/images/shirt4.png'}
+                onClick={() => {
+                  props.setProductImage('/images/shirt4.png');
+                }}
+              ></ProductColorBox>
+              <ProductColorBox
+                picture={'/images/shirt5.png'}
+                onClick={() => {
+                  props.setProductImage('/images/shirt5.png');
+                }}
+              ></ProductColorBox>
+              <ProductColorBox
+                picture={'/images/shirt6.png'}
+                onClick={() => {
+                  props.setProductImage('/images/shirt6.png');
+                }}
+              ></ProductColorBox>
+            </ul>
+          </ProductColor>
+        </OptionChoice>
+
+        <Custom
+          setLocation={setLocation}
+          setFontStyle={props.setFontStyle}
+          setFontColor={props.setFontColor}
+          eventHandler={props.eventHandler}
+          location={location}
+        />
+
+        <CartContainer />
+      </DetailBox>
+    </div>
+  );
+};
+
+let DetailBox = styled.div`
+  flex: 0 0 41.66667%;
+  max-width: 41.66667%;
+  width: 900px;
+  padding: 0 24px 0 8px;
+  background-color: yellow;
+`;
+
+let DetailTitle = styled.div`
+  display: flex;
+  align-items: center;
+  margin: 14px 0 4px 0;
+  height: 38px;
+  font-size: 32px;
+  font-weight: 500;
+`;
+let DetailPrice = styled.div`
+  display: flex;
+  align-items: center;
+  margin: 20px 0 8px 0;
+  height: 22px;
+  font-size: 22px;
+  font-weight: bolder;
+`;
+let DetailRate = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 24px;
+  height: 18px;
+`;
+
+let OptionChoice = styled.div`
+  margin: 8px 0;
+  background-color: white;
+`;
+
+let OptionTilte = styled.div`
+  padding: 15px 3px 5px 3px;
+  font-size: 15px;
+`;
+
+let DropDown = styled.ul`
+  margin: 3px;
+  width: 97.5%;
+  height: 48px;
+  color: #535353;
+  border-radius: 4px;
+  border: 1px solid black;
+  cursor: pointer;
+  li {
+    display: flex;
+    padding: 18px;
+  }
+`;
+
+let DropDownMenu = styled.ul`
+  position: relative;
+  margin-top: -5px;
+  height: 48px;
+  border-bottom: 1px solid #ccc;
+  li {
+    width: 90%;
+    background-color: white;
+    border-radius: 4px;
+    border: 1px solid black;
+    color: #535353;
+  }
+  li:hover {
+    background-color: #535353;
+    color: white;
+  }
+`;
+
+let ProductColor = styled.div`
+  ul {
+    display: flex;
+    flex-wrap: wrap;
+  }
+`;
```

Comment:
```suggestion
let ProductColor = styled.div`
  ul {
    display: flex;
    flex-wrap: wrap;
  }
`;
```
이럴거면 UI을 스타일드 컴포넌트로 선언하는게 좋지 않을까요? 네스팅을 위해서 스타일드 컴포넌트를 사용하는 방식은 좋지 않습니다.

Reply:
넵 확인했습니다